components/board/genericlinux/i2c.go

```diff
@@ -0,0 +1,125 @@
+package genericlinux
+
+import (
+	"context"
+	"encoding/binary"
+	"sync"
+
+	"github.com/edaniels/golog"
+	"periph.io/x/conn/v3/i2c"
+	"periph.io/x/conn/v3/i2c/i2creg"
+	"periph.io/x/host/v3"
+
+	"go.viam.com/rdk/components/board"
+)
+
+func init() {
+	if _, err := host.Init(); err != nil {
+		golog.Global().Debugw("error initializing host", "error", err)
+	}
+}
+
+type i2cBus struct {
+	internal i2c.BusCloser
+	mu       sync.Mutex
+}
+
+func newI2cBus(deviceName string) (*i2cBus, error) {
+	// We return a pointer to an i2cBus instead of an i2cBus itself so that we can return nil if
+	// something goes wrong.
+	bus, err := i2creg.Open(deviceName)
+	if err != nil {
+		return nil, err
+	}
+	return &i2cBus{internal: bus}, nil
+}
+
+// This lets the i2cBus type implement the board.I2C interface.
+func (bus *i2cBus) OpenHandle(addr byte) (board.I2CHandle, error) {
+	bus.mu.Lock() // Lock the bus so no other handle can use it until this one is closed.
+	return &i2cHandle{device: &i2c.Dev{Bus: bus.internal, Addr: uint16(addr)}, mu: &bus.mu}, nil
+}
+
+type i2cHandle struct {
+	device *i2c.Dev    // Will become nil if we close the connection
+	mu     *sync.Mutex // Points to the I2C bus' mutex
```

Comment:
Make the comments match up better with the names in the code: "...if we Close() the connection" and "the i2cBus' mutex"